File: backend/src/main/java/com/example/backend/Model/Class/PartProduction.java
package com.example.backend.Model.Class;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Cacheable
@Data
@Entity
@Table(name = "PART_PRODUCTION")
public class PartProduction {
    @Id
    private String id;
    @Column(name = "produced_date")
    private LocalDateTime produced_date;
    @Column(name = "quantity")
    private Integer quantity;
    @ManyToOne
    @JoinColumn(name = "machine_id",referencedColumnName = "id")
    private Machines machine_id;
    @ManyToOne
    @JoinColumn(name = "part_id",referencedColumnName = "id")
    private Parts part_id;
    public static final String QUERY = " FROM PartProduction pp WHERE (LOWER(pp.machine_id.name) = " +
            "LOWER(:machine_name_or_id) OR LOWER(pp.machine_id.id) = LOWER(:machine_name_or_id)) ";
}

File: frontend/src/app/_model/_interface/part-production.ts
import { isMachine, Machines } from './machine';
import { isParts, Parts } from './parts';

export interface PartProduction {
  id: string;
  produced_date: string;
  quantity: number;
  machine_id: Machines;
  part_id: Parts;
}

export function isPartProduction(obj: any): obj is PartProduction {
  return (
    obj &&
    typeof obj.id === 'string' &&
    typeof obj.produced_date === 'string' &&
    typeof obj.quantity === 'number' &&
    isMachine(obj.machine_id) &&
    isParts(obj.part_id)
  );
}
